client/taskenv/env.go

```diff
@@ -48,9 +48,15 @@ const (
 	// GroupName is the environment variable for passing the task group name.
 	GroupName = "NOMAD_GROUP_NAME"
 
+	// JobID is the environment variable for passing the job ID.
+	JobID = "NOMAD_JOB_ID"
+
 	// JobName is the environment variable for passing the job name.
 	JobName = "NOMAD_JOB_NAME"
 
+	// JobVersion is the environment variable for passing the job version.
+	JobVersion = "NOMAD_JOB_VERSION"
```

Comment:
Although I definitely understand why you'd want the job version to be exposed as an environment variable, unfortunately I think it's misleading to have.

A job can be updated in-place without causing existing allocations to be replaced, which means those allocations will have the wrong `NOMAD_JOB_VERSION` environment variable. For example if I have a job with a group `count = 2` and I change it to `count = 3`, I get a new allocation but my existing allocations are unchanged.

Reply:
Thank you for pointing that behavior out - I was not aware of it. Would you have a proposal on how to achieve this instead?
For my use case, changes that do not change the code or config (i.e. other env variables) are not relevant and so I would be fine with this behavior as-is. If there is no straightforward way to improve, do you think renaming the variable and adding documentation could mitigate the risk here?

Reply:
The best way to accomplish this is to set a [`meta`](https://www.nomadproject.io/docs/job-specification/meta) field. I've used this in production cases to provide a consistent "version" for the job across deployments (ex. `v1.2.3-abcdef` or whatever).